HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,706 @@
+
+### Zero-Balance Accounts
+
+Accounts with zero HBAR balance must be able to receive tokens. This can be done in the following ways:
+
+- A sender may send a token to an unused EVM address (alias) and the account will be auto-created without any HBAR balance. It will also be created with `max_auto_associations` set to `-1`, permitting unlimited auto-associations. Subsequent transfers of any token to this account with no hbars will succeed.
+- A wallet or other entity can create an account with `max_auto_associations` set to `-1`. Tokens can then be transferred into the account without the account owning any hbars.
+- An existing account can change `max_auto_associations` to `-1`, and then subsequent token transfers into the account do not charge the account.
+- An account with no open auto-association slots can receive airdrops but must send a `TokenClaimAirdropTransaction`, which requires a payer. If the account has zero hbars, then it can still claim the transfer if someone else is willing to pay for that transaction. For example, a Dapp could be the payer on the transaction. Both the Dapp and the account must sign the transaction.
+
+### Token Reject
+
+Today, if an account holds no balance (or no serial) of a token type, then the account can be dissociated from the token. However, tokens may have hefty custom fees for transferring a token. A user may not be able or willing to pay those fees, but cannot dissociate without doing so, effectively forcing them to pay for auto-renewal of tokens they do not want.
+
+This can be a significant security concern. A token with custom fees could  be used by a token spammer to airdrop a token to all users with a custom fee that will charge the users some HBAR or other tokens in order to get rid of the token — essentially creating a “ransom” token. To prevent this, we introduce the new TokenRejectTransaction.
+
+Any token (fungible or non-fungible) can be rejected, whether it was created through automatic or manual association, and no matter how long the token was held. A rejected token is transferred back to the token treasury, and all custom fees on the token are waived.
+
+> ⚠️ When a token is rejected, we **do not** decrement `used_auto_associations`. That field tracks the total number of times auto-associations have been made, not the current number of token types on the account that were auto-associated.
+
+
+> ⚠️ `receiver_sig_required` is ignored on the treasury account when the `TokenReject` is handled. Reject will always work with no assessed custom fees.
```

Comment:
if this is "correcting" a problem with existing behavior regarding auto associated ransom tokens (we warned you) ought this be a different HIP?  This hip is about airdrops.

Reply:
@david-bakin-sl we're updating the HAPI spec (found in the protobuf comments) with the cumulative set of details like this. The HIP just defines a particular feature, the sum total should be in the protobuf definitions.

Reply:
@bugbytesinc This HIP needs the reject transaction to be complete otherwise it makes the current situation much worse. It could be a separate HIP, but then airdrops cannot go out without reject being implemented. So having them together seemed sensible.

Reply:
@rbair23  - ok, that's fine with me ... and I'd hope there's some idea that in the future sometime we can have a pbj/javadoc hybrid that pulls some kind of documentation out of there?  I.e., maybe there'd be mark-up tags that call out behavior vs security checks vs I dunno, other stuff you'd like to see summarized?  I'm not sure what this would look like ... but I'm also not aware that anywhere over in docs.hedera.com now that says anything like you should check the protobufs for the actual enforced semantics.  (Considering most people will use an SDK which wraps all the protobufs and HAPI calls and thus wouldn't really think of looking in the protobuf sources for the authoritative rules.)